### PR TITLE
[DOCS] Reorganize documentation

### PR DESCRIPTION
**Pull request by mattcieslak**
QSIRecon needed a major overhaul of the documentation, especially for the recently-added mapping workflows.

### CODE REVIEW

File: docs/building_workflows.rst [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      "name": "track_ifod2",
+      "software": "MRTrix3",
+      "action": "tractography",
+      "qsirecon_suffix": "MRtrix3_act-FAST",
+      "input": "msmt_csd"
+    },
+
+NOTE: there can only be zero (inputs come from the input data) or one
+(inputs come from a mixture of the input data and the "input" node)
+name specified for ``"input"``.
```
Comment (tsalo) [inline docs/building_workflows.rst:119]: Good opportunity for an admonition!

```suggestion
.. note::
    There can only be zero (inputs come from the input data) or one
    (inputs come from a mixture of the input data and the "input" node)
    name specified for ``"input"``.
```

File: docs/changes.rst
Comment (tsalo) [inline docs/changes.rst:1]: How do you feel about copying over the release notes before making each release, like I do with XCP-D? https://xcp-d.readthedocs.io/en/latest/changes.html

File: docs/building_workflows.rst [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+
+
+A *node* in the QSIRecon ``"nodes"`` list represents a unit of processing.
+Unlike NiPype nodes, which represent a single call to a commandline program
+or function, a node here represents a set of steps that accomplish a task.
+For example, if you want to perform CSD there are other steps that should
+happen before actually fitting spherical harmonics to the data. The entry
+for this in the ``"nodes"`` list could look like:
+
```
Comment (tsalo) [inline docs/building_workflows.rst:55]: One small thing- I like to put each sentence on a separate line, sometimes split on punctuation if it gets too long. I think that leads to cleaner diffs in the long run.

File: docs/reconstruction copy.rst
Comment (tsalo) [inline docs/reconstruction copy.rst:1]: Maybe delete this file?

File: docs/reconstruction.rst
Comment (tsalo) [inline docs/reconstruction.rst:1]: Consider applying the changes from https://github.com/PennLINC/qsiprep/pull/801 here. I think it's just changing `--recon_spec` to `--recon-spec` in the CLI call examples.

File: docs/license.rst [hunk 3]
```diff
@@ -7,5 +7,5 @@ in the ``qsirecon`` distribution.
 
 All trademarks referenced herein are property of their respective holders.
 
-Copyright (c) 2018-2019 PennLINC
+Copyright (c) 2018-2024 PennLINC
```
Comment (tsalo) [inline docs/license.rst:10]: I recall something about not needing to change end-dates for licenses. I think you can just have `2018`.

File: docs/changes.rst
Comment (mattcieslak) [reply docs/changes.rst:1]: In the past I've had trouble keeping this in sync with the github releases page. Do you have a system for keeping these all consistent?

Comment (tsalo) [reply docs/changes.rst:1]: Here's my workflow:

1. Use the "Generate Release Notes" button on a draft release on GitHub.
2. Add some context to the release notes.
3. Save the release as a draft.
4. Copy the release notes to the CHANGES.rst file and update the CITATION.cff file in a new PR.
5. Merge the PR.
6. Make the new release.

It's a little annoying, but it gets the release notes into the CHANGES file.

**Pull request merged**